Dragonflight/WarriorArms.lua

```diff
@@ -852,29 +913,34 @@ spec:RegisterAbilities( {
     execute = {
         id = 163201,
         cast = 0,
-        cooldown = function () return (talent.improved_execute.enabled and 0 or 6) end,
+        cooldown = function () return ( talent.improved_execute.enabled and 0 or 6 ) end,
         gcd = "spell",
+        hasteCD = true,
         noOverride = 317485, -- Condemn
 
-        spend = function () return min(max(rage.current, 20),40) end,
+        spend = function ()
```

Comment:
This is going to cause some problematic behavior.  If you have the actual rage expenditure here, it will be spent before the handler runs.  See the Shadowlands implementation of Execute to make sure we don't lose functionality.

Reply:
CS fixed in https://github.com/Hekili/hekili/pull/1670/commits/33370bc31a64a614f1d30da0c18027f72e0beb2f
WB fixed in https://github.com/Hekili/hekili/pull/1670/commits/4cc68109cde878a259f4b46c5c1ad90194cca215

Reply:
Refactored Arms in https://github.com/Hekili/hekili/pull/1670/commits/4842caaeb90084100fe9eaa685f7c40a822cf01f

Reply:
Refactored all executes for all specs in https://github.com/Hekili/hekili/pull/1670/commits/fcc7514052a4e3f5469edcddf0ce4a01d944e32b to work better.

Please re-review all 3 specs executes as they are obviously tricky.